.github/workflows/zxc-single-day-longevity-test.yaml

```
@@ -685,7 +706,7 @@ jobs:
   single-day-longevity-test-result:
     name: Calculate Result
     runs-on: hiero-citr-linux-large
-    if: ${{ always() }}
+    if: ${{ !cancelled() }}
```

```suggestion
    if: ${{ !cancelled() && always() }}
```

I think you need to specify `&& always()` here to tell the step to run if previous steps failed.